Story/user screen2

ksiegel1923 opened the pull request.

ksiegel1923 committed:
user screen 2 component built, error with passing onCheck function through nested components

ksiegel1923 committed:
content from forms on second screen saved on the press of the submit button in two separate objects

ksiegel1923 committed:
refactored module component so screen structure is more similar to how frame131 was set up

michellexliu commented on feedback-module/src/components/pages/Screen2.js:
```diff
@@ -8,34 +8,66 @@ import ModuleButton from "../common/Button";
         [label]: false,
       };
     });
     setCheckedFields(fields);
```
I noticed that you also setCheckedFields in the onCheck and pass in setInputQuestions as a prop in TextBoxList— is it still necessary to have this useEffect?

ksiegel1923 replied:
The onCheck function assumes that all the values are already in the checkedFields object (set to false) and the TextBoxList takes in the InputQuestions as a prop (also assuming it is already populated). This useEffect initially populates all the objects. I could remove it but it would require altering the other functions (I didn't think I would want to reset all the values every time a checkbox was checked or there was an onChange (I think I only need to set this once which is why I thought of putting it in the useEffect)

michellexliu replied:
Hmmm would it be possible to populate the objects when setting useState since we're only trying to do this for the initial render?

So instead of `useState({})` we could maybe do something like
```
useState(
{feedbackType.checkboxes.map((label) => {
      fields = {
        ...fields,
        [label]: false,
      };
    });}
)
```

ksiegel1923 replied:
Yeah I think a useEffect with an empty dependency array has the same effect but if you think this is cleaner then I can change it

ksiegel1923 replied:
Not sure how I would populate the array inside the useState

michellexliu commented on feedback-module/src/components/pages/Screen2.js:
```diff
@@ -8,34 +8,66 @@ import ModuleButton from "../common/Button";
 
 function Screen2({ feedbackType }) {
   const [checkedFields, setCheckedFields] = useState({});
```
Since we're storing the checkedFields in objects and using the labels as keys, I'm wondering how this will affect us in the future when we're parsing the data to be sent back to the webmasters. Because different forms will have different labels & thus different keys, it seems like it would be more challenging to do things like filtering the data or formatting than in an array for example.

ksiegel1923 replied:
Interesting I could see how this might present challenges but how would we format it in an array because don't we want to know which checkboxes have been checked and which haven't (this is the question I was asking you about yesterday morning when I was setting up the data structure)

ksiegel1923 replied:
I was thinking an object could work because we could just send this object to the backend and it will be very clear what questions were asked and what the responses were (the recipient of the email doesn't need to know the questions that the user didn't see I don't think)

michellexliu replied:
I was thinking that instead of the array containing just the checked elements, we can map through the checkbox labels and create an array of objects with the format `{label: "checkbox label", checked: false}`. Then for the `onCheck` function, instead of passing in the label, we can pass in the index.

michellexliu commented on feedback-module/src/components/pages/Screen2.js:
```diff
@@ -8,34 +8,66 @@ import ModuleButton from "../common/Button";
-        <CheckboxList feedbackType={feedbackType} onCheck={() => onCheck} />
-        <TextboxList feedbackType={feedbackType} />
+        <CheckboxList
+          feedbackType={feedbackType}
```
Minor point, but would it be possible to just directly pass in feedbackType.checkboxes? It doesn't seem like you're using any other part of feedbackType within the CheckboxList component

ksiegel1923 replied:
Yes definitely good point, i'll fix that now

ksiegel1923 replied:
I also changed it for the textboxList as well

ksiegel1923 committed:
implemented PR comments:
only send info about checkboxes particularly to checkbox component rather than all the questions
change structure of info saved from object to an array of objects

michellexliu commented on feedback-module/src/components/pages/Screen2.js:
```diff
@@ -6,43 +6,52 @@ import TextboxList from "../TextboxList";
+        checkedField.label === "Other" &&
+        checkedField.checked &&
+        checked.push({
+          label: "Other text",
```
For the other case, instead of pushing a new object entirely, could we just set an info field for the pre-existing "other" check element?

So instead of `checked.push(....)` we could just do `checkedField.info=otherField`

ksiegel1923 committed:
implemented PR comments:
remove useEffect and just set info in the useState
update the other object so it adds a field info if the user added more specific info

ksiegel1923 committed:
fixed merge conflicts and resolved console error with form embedded inside another form
